dht-node/src/dht_node/index.ts

```diff
@@ -184,3 +187,38 @@ export const startDHT = async (topic: string): Promise<void> => {
     logger.error('DHT unexpected error:', err)
   }
 }
+
+async function writeHomeCommunityEnries(pubKey: any): Promise<CommunityApi[]> {
+  const homeApiVersions: CommunityApi[] = Object.values(ApiVersionType).map(function (apiEnum) {
+    const port =
+      Number.parseInt(CONFIG.FEDERATION_COMMUNITY_API_PORT) + Number(apiEnum.replace('_', ''))
+    const comApi: CommunityApi = {
+      api: apiEnum,
+      url: CONFIG.FEDERATION_COMMUNITY_URL + ':' + port.toString() + '/api/',
+    }
+    return comApi
+  })
+  try {
+    // first remove privious existing homeCommunity entries
```

Comment:
You don't remove the homeCommunity here but the foreignCommunities

Reply:
why do you think, that instead of home-community-entries the foreign-community-entries are removed?
the home-community-entries have the foreign-flag = false.